pkg/operator/nfs/controller.go

```diff
@@ -0,0 +1,480 @@
+func NewController(context *clusterd.Context, containerImage string) *Controller {
+	return &Controller{
+		context:        context,
+		containerImage: containerImage,
+	}
+}
+
+// StartWatch watches for instances of nfsserver custom resources and acts on them
+func (c *Controller) StartWatch(namespace string, stopCh chan struct{}) error {
+	resourceHandlerFuncs := cache.ResourceEventHandlerFuncs{
+		AddFunc:    c.onAdd,
+		UpdateFunc: c.onUpdate,
+		DeleteFunc: c.onDelete,
+	}
+
+	logger.Infof("start watching nfs server resources in namespace %s", namespace)
+	watcher := opkit.NewWatcher(NFSResource, namespace, resourceHandlerFuncs, c.context.RookClientset.NfsV1alpha1().RESTClient())
+	go watcher.Watch(&nfsv1alpha1.NFSServer{}, stopCh)
+
+	return nil
+}
+
+type nfsServer struct {
+	name      string
+	context   *clusterd.Context
+	namespace string
+	spec      nfsv1alpha1.NFSServerSpec
+	ownerRef  metav1.OwnerReference
+}
+
+func newNfsServer(c *nfsv1alpha1.NFSServer, context *clusterd.Context) *nfsServer {
+	return &nfsServer{
+		name:      appName,
+		context:   context,
+		namespace: c.Namespace,
+		spec:      c.Spec,
+		ownerRef:  nfsOwnerRef(c.Namespace, string(c.UID)),
+	}
+}
+
+func nfsOwnerRef(namespace, nfsServerID string) metav1.OwnerReference {
+	blockOwner := true
+	return metav1.OwnerReference{
+		APIVersion:         NFSResource.Version,
+		Kind:               NFSResource.Kind,
+		Name:               namespace,
+		UID:                types.UID(nfsServerID),
+		BlockOwnerDeletion: &blockOwner,
+	}
+}
+
+func getServerConfig(exports []nfsv1alpha1.ExportsSpec) map[string]string {
+	configOpt := make(map[string]string)
+
+	for _, export := range exports {
+		configOpt["accessMode"] = export.Server.AccessMode
+		configOpt["squash"] = export.Server.Squash
+	}
+
+	return configOpt
+}
+
+func getClientConfig(exports []nfsv1alpha1.ExportsSpec) map[string]string {
+	configOpt := make(map[string]string)
+
+	for _, export := range exports {
+		for _, client := range export.Server.AllowedClients {
+			configOpt["accessMode"] = client.AccessMode
+			configOpt["squash"] = client.Squash
+		}
+	}
+
+	return configOpt
+}
+
+func validateServerConfig(spec *nfsv1alpha1.NFSServerSpec) error {
+	serverConfig := getServerConfig(spec.Exports)
+	clientConfig := getClientConfig(spec.Exports)
+
+	accessMode := serverConfig["accessMode"]
+	squash := serverConfig["squash"]
+	clientAccessMode := clientConfig["accessMode"]
+	clientSquash := clientConfig["squash"]
+
+	if err := checkAccessMode(accessMode); err != nil {
+		return err
+	}
+
+	if err := checkAccessMode(clientAccessMode); err != nil {
+		return err
+	}
+
+	if err := checkSquash(squash); err != nil {
+		return err
+	}
+
+	if err := checkSquash(clientSquash); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func checkAccessMode(accessMode string) error {
+	// TODO: Add code to check access mode for "ReadWrite", "ReadOnly" and "none"
```

Comment:
Instead of commenting out the unused code, i would recommend just removing the unimplemented function.